framework/src/modules/token/method.ts

```diff
@@ -734,4 +799,24 @@ export class TokenMethod extends BaseMethod {
 			.get(EscrowStore)
 			.has(methodContext, this.stores.get(EscrowStore).getKey(chainID, tokenID));
 	}
+
+	private async _isAllTokensSupported(methodContext: MethodContext): Promise<boolean> {
+		const supportedTokensStore = this.stores.get(SupportedTokensStore);
+
+		return supportedTokensStore.has(methodContext, ALL_SUPPORTED_TOKENS_KEY);
+	}
+
+	private async _getSupportedTokens(
+		methodContext: MethodContext,
+		chainID: Buffer,
+	): Promise<SupportedTokensStoreData> {
+		const supportedTokensStore = this.stores.get(SupportedTokensStore);
+		const supportedTokens = await supportedTokensStore.get(methodContext, chainID);
+
+		if (!supportedTokens) {
+			throw new Error('No supported tokens found.');
+		}
```

Comment:
as mentioned before, `await supportedTokensStore.get` will throw if `chainID` doesn't exist

Reply:
this is still not addressed